migrations/20201111_api_date.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Fix dates in the database that were stored as $type string instead of Date
+ */
+/* eslint-disable no-console */
+
+const MIGRATION_NAME = '20201111_api_date';
+
+import * as Tasks from '../website/server/models/task';
+
+const progressCount = 1000;
+let count = 0;
+
+async function updateUser (user) {
+  count++;
+
+  let set = {};
+  let newDate = new Date(user.date);
+  if(isValidDate(newDate)) {
+    set = {
+      'date': newDate
+    }
+  } else {
+    set = {
+      'date': ''
+    }
+  }
+
+  set.migration = MIGRATION_NAME;
+
+  if (count % progressCount === 0) console.warn(`${count} ${user._id}`);
+  return await Tasks.Task.update({_id: user._id}, {$set: set}).exec();
+}
+
+module.exports = async function processUsers () {
+  let query = {
```

Comment:
this query will find all tasks and unfortunately that would take ages to finish as we have more than 60 milion ones, can we restrict it so that only the affected tasks (those with a date field using mongo's $exist parameter) are targeted?

Reply:
also migration can be removed as it's only used for users